dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -489,13 +497,44 @@ class E2EPerfTest extends PerfTest {
     String testDirectory,
     String testTarget, {
     String summaryFilename,
+    String testDriver,
     List<String> benchmarkScoreKeys,
     }
   ) : super(
     testDirectory,
     testTarget,
     summaryFilename,
-    testDriver: 'test_driver/e2e_test.dart',
+    testDriver: testDriver?? 'test_driver/e2e_test.dart',
+    needsFullTimeline: false,
+    benchmarkScoreKeys: benchmarkScoreKeys ?? const <String>[
+      'average_frame_build_time_millis',
+      'worst_frame_build_time_millis',
+      '90th_percentile_frame_build_time_millis',
+      '99th_percentile_frame_build_time_millis',
+      'average_frame_rasterizer_time_millis',
+      'worst_frame_rasterizer_time_millis',
+      '90th_percentile_frame_rasterizer_time_millis',
+      '99th_percentile_frame_rasterizer_time_millis',
+      ],
+  );
+
+  @override
+  String get resultFilename => timelineFileName ?? 'e2e_perf_summary';
+}
+
+class E2EPerfTestWithSkSL extends PerfTestWithSkSL {
```

Comment:
I'm not very satisfied with this part because it's almost duplicating `E2EPerfTest` except for what class it inherits. But I don't know if there's better way of implementing diamond inheritance.